### PR TITLE
compress binary while sending it to all the nodes

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
compress binary while sending it to all the nodes

## Motivation and Context
Also, limit the amount of concurrency when sending
binary updates to peers, avoid high network over
TX that can cause disconnection events for the
node sending updates.

## How to test this PR?
Just build minio with a RELEASE date that is older than
the current RELEASE date, you get to use the `mc admin update alias/`
command.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/update.go [hunk 1]
```diff
@@ -510,20 +512,37 @@ func getUpdateReaderFromURL(u *url.URL, transport http.RoundTripper, mode string
...
+	bc := bytebufferpool.Get()
+	defer func() {
+		b.Reset()
+		bc.Reset()
+
+		bytebufferpool.Put(b)
+		bytebufferpool.Put(bc)
+	}()
+
+	w := pgzip.NewWriter(bc)
```
Comment (klauspost) [inline cmd/update.go]:
```suggestion
	w := zstd.NewWriter(bc)
```

File: cmd/peer-rest-server.go [hunk 2]
```diff
@@ -850,7 +851,14 @@ func (s *peerRESTServer) VerifyBinaryHandler(w http.ResponseWriter, r *http.Requ
 	}
 	releaseInfo := r.Form.Get(peerRESTReleaseInfo)
 
-	if err = verifyBinary(u, sha256Sum, releaseInfo, getMinioMode(), r.Body); err != nil {
+	pr, err := pgzip.NewReader(r.Body)
```
Comment (klauspost) [inline cmd/peer-rest-server.go]:
```suggestion
	pr, err := zstd.NewReader(r.Body)
```

File: cmd/update.go [hunk 3]
```diff
@@ -510,20 +512,37 @@ func getUpdateReaderFromURL(u *url.URL, transport http.RoundTripper, mode string
...
+	if _, err = io.Copy(w, io.TeeReader(reader, b)); err != nil {
+		return nil, nil, err
+	}
+
+	w.Flush()
```
Comment (klauspost) [inline cmd/update.go]: Close flushes (on both)
```suggestion
```

**Review by klauspost — COMMENTED** [review]
Not a big difference (~2MB), but we might as well take it.

(imports should be changed ofc)

**Comment by harshavardhana** [comment]
done @klauspost

**Review by pjuarezd — APPROVED** [review]

**Review by kannappanr — APPROVED** [review]
lgtm

**Pull request merged**